app/main.py

```diff
@@ -1,2 +1,81 @@
 class Dictionary:
-    pass
+    def __init__(self):
+        self.hash_table = [None] * 8
+
+    def get_need_index(self, ind, key, hash_key):
+        if self.hash_table[ind] is None:
+            return ind
+        if self.hash_table[ind][1] == hash_key \
+                and self.hash_table[ind][0] == key:
+            return ind
+
+        # collision
+        need_item = [
```

Comment:
This algorithm is too slow. Try to use open addressing

Reply:
You haven't fixed this algorithm yet. It works but it works slowly

Reply:
You should avoid using such comprehensions.

Reply:
They iterate over all items which leads to the increasing the time complexity
Ask in slack if you have questions about it.